app/Foliage/CmdBuild.hs

```diff
@@ -320,8 +320,14 @@ getExtraEntries packageVersions =
           -- Calculate (by applying them chronologically) the effective `VersionRange` for the package group.
           effectiveRanges :: [(UTCTime, VersionRange)]
           effectiveRanges = NE.tail $ NE.scanl applyChangeToRange (posixSecondsToUTCTime 0, anyVersion) deprecationChanges
+
           -- Create a `Tar.Entry` for the package group, its computed `VersionRange` and a timestamp.
-          createTarEntry (ts, effectiveRange) = mkTarEntry (BL.pack $ prettyShow effectiveRange) (IndexPkgPrefs pn) ts
+          createTarEntry (ts, effectiveRange) = mkTarEntry prettyEffectiveRange (IndexPkgPrefs pn) ts
+            where
+              -- We have to prepend the package name despite this file residing
+              -- in a package-specific directory.
+              prettyEffectiveRange =
+                BL.pack $ unPackageName pn <> " " <> prettyShow effectiveRange
```

Comment:
Since you tracked down that cabal parses this as a `Dependency` I suggest we pretty-print it as a `Dependency` too, so to avoid any incompatibility.

```suggestion
          createTarEntry (ts, effectiveRange) = mkTarEntry (BL.pack $ prettyShow dep) (IndexPkgPrefs pn) ts
            where
              dep = mkDependency pn effectiveRange mainLibSet
```

Where `mkDependency` and `mainLibSet` are defined in `Distribution.Types.Dependency`.

Reply:
Yeah, that makes sense; I still find `Dependency` a bit misleading in this specific context as sublibraries are not applicable here (Cabal will simply ignore them, see [here](https://github.com/haskell/cabal/blob/a191f0a2bf72894b3d830ca3ee4ea111d96b6f65/cabal-install/src/Distribution/Client/IndexUtils.hs#L406)), but I will just add a comment :+1:

Reply:
Also confirming that this works just as well as the previous approach

Reply:
> Yeah, that makes sense; I still find `Dependency` a bit misleading in this specific context as sublibraries are not applicable here

100% agree with you. This also made me just realise per-component solving won't work with deprecated-packages OOTB (since they only deprecate the main lib and the solver could still pick some other lib). This should be reworked in cabal-install ... (one day :roll_eyes:)